aptos-move/framework/move-stdlib/sources/vector.move

```diff
@@ -122,6 +122,22 @@ module std::vector {
         pragma intrinsic = true;
     }
 
+    /// Splits the collection into two at the given index.
+    /// Returns a newly allocated vector containing the elements in the range [at, len).
+    /// After the call, the original vector will be left containing the elements [0, at)
+    /// with its previous capacity unchanged.
+    public fun split_off<Element>(self: &mut vector<Element>, at: u64): vector<Element> {
```

Comment:
what's the diff with `trim`?

Reply:
I believe it's better to make trim native tho

Reply:
the move vector seems not to have the concept of `capacity`... 
idk who added this... but `push_back` one by one is not that efficient? :)

Reply:
I did that in #14862, let me just merge this into that PR.